tiledb/api/src/array/mod.rs

```diff
@@ -380,6 +382,113 @@ impl Array {
         Ok(Some(datatype))
     }
 
+    pub fn vacuum<S>(
+        ctx: &Context,
+        array_uri: S,
+        config: &Config,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let c_array_uri = cstring!(array_uri.as_ref());
+        ctx.capi_call(|ctx| unsafe {
+            ffi::tiledb_array_vacuum(ctx, c_array_uri.as_ptr(), config.capi())
+        })?;
+        Ok(())
+    }
+
+    pub fn upgrade_version<S>(
+        ctx: &Context,
+        array_uri: S,
+        config: &Config,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let c_array_uri = cstring!(array_uri.as_ref());
+        ctx.capi_call(|ctx| unsafe {
+            ffi::tiledb_array_upgrade_version(
+                ctx,
+                c_array_uri.as_ptr(),
+                config.capi(),
+            )
+        })?;
+        Ok(())
+    }
+
+    pub fn consolidate<S>(
+        ctx: &Context,
+        array_uri: S,
+        config: &Config,
+    ) -> TileDBResult<()>
+    where
+        S: AsRef<str>,
+    {
+        let c_array_uri = cstring!(array_uri.as_ref());
+        ctx.capi_call(|ctx| unsafe {
+            ffi::tiledb_array_consolidate(
+                ctx,
+                c_array_uri.as_ptr(),
+                config.capi(),
+            )
+        })?;
+        Ok(())
+    }
+
+    pub fn consolidate_fragments<S>(
```

Comment:
TODO: test consolidate fragments and consolidate fragments with key.

Reply:
Also perhaps split those for a future pr?

Reply:
I'd recommend adding them to this one just so we don't forget.  Stories for "add tests to thing" tend to languish, in my experience.

Reply:
done